### PR TITLE
added first implementation of utils sub-module

### PR DESCRIPTION
**Pull request by v923z**
This PR should address the issue raised in https://github.com/v923z/micropython-ulab/issues/341

The general problem is that peripheral devices do not necessarily supply data compatible with `ulab`'s 5 supported `dtypes`. However, for the most common tasks, simple conversion functions can be added. With the help of these, arbitrary data formats could be converted to the supporetd `dtype`s. After the conversion, the standard `ulab` operations can be carried out on the data arrays.

In the future, the `utils` sub-module could be the place, where useful, but not `numpy`-compatible functions working with or returning `ndarray`s could be stored.

### CODE REVIEW

File: code/utils/utils.c [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+                    memcpy(--tmpbuff, buffer++, 1);
+                }
+                if(buffer_type == UTILS_INT32_BUFFER) {
+                    *array++ = (mp_float_t)(*(int32_t *)tmpbuff);
+                } else {
+                    *array++ = (mp_float_t)(*(uint32_t *)tmpbuff);
+                }
+            }
+        } else {
+            for(uint8_t i = 0; i < len; i++) {
```
Comment (pumelo) [inline code/utils/utils.c:92]: use size_t? uint8_t will probably not reach the end condition...

**Review by pumelo — CHANGES_REQUESTED** [review]
This is probably the cause for the exception I saw.. About to recompile and test .. 🏃

File: code/utils/utils.c (hunk 1 above)
Comment (pumelo) [reply code/utils/utils.c:92]: Yes, this was the issue! with the documented crash.

Comment (v923z) [reply code/utils/utils.c]: Indeed! 😊 Thanks for catching this!

**Comment by v923z** [comment]
@ubIQio  This can also solve your issue https://github.com/v923z/micropython-ulab/issues/306

**Pull request merged**